txn.go

```diff
@@ -33,9 +33,23 @@ type Txn struct {
 	rootTxn *iradix.Txn
 	after   []func()
 
+	// changeSet is used to track the changes performed during the transaction. If
+	// it is nil at transaction start then changes are not tracked.
+	changeSet ChangeSet
+
 	modified map[tableIndex]*iradix.Txn
 }
 
+// TrackChanges enables change tracking for the transaction. If called at any
+// point before commit, subsequent mutations will be recorded and can be
+// retrieved using ChangeSet. Once this has been called on a transaction it
+// can't be unset.
+func (txn *Txn) TrackChanges() {
```

Comment:
I would add a note about how to call this function similar to `ChangeSet()`:

> // As with other Txn methods
> // it's not safe to call this from a different goroutine than the one making
// mutations or committing the transaction.